app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.test.js

```diff
@@ -16,125 +16,238 @@ const MOCK_MAINNET_CONFIGURATION = {
   type: NETWORK_TYPES.MAINNET,
 };
 const MOCK_LINEA_MAINNET_CONFIGURATION = {
-  id: 123,
+  id: 1234,
   chainId: CHAIN_IDS.LINEA_MAINNET,
   type: NETWORK_TYPES.LINEA_MAINNET,
 };
 
 describe('switchEthereumChainHandler', () => {
-  it('should call setProviderType when switching to a built in infura network', async () => {
-    const mockSetProviderType = jest.fn();
-    const mockSetActiveNetwork = jest.fn();
-    const switchEthereumChainHandler = switchEthereumChain.implementation;
-    await switchEthereumChainHandler(
-      {
-        origin: 'example.com',
-        params: [{ chainId: CHAIN_IDS.MAINNET }],
-      },
-      {},
-      jest.fn(),
-      jest.fn(),
-      {
-        getCurrentChainId: () => NON_INFURA_CHAIN_ID,
-        findNetworkClientIdByChainId: () => 123,
-        setNetworkClientIdForDomain: jest.fn(),
-        getProviderConfig: () => ({ chainId: NON_INFURA_CHAIN_ID }),
-        findNetworkConfigurationBy: () => MOCK_MAINNET_CONFIGURATION,
-        setProviderType: mockSetProviderType,
-        setActiveNetwork: mockSetActiveNetwork,
-        requestUserApproval: mockRequestUserApproval,
-      },
-    );
-    expect(mockSetProviderType).toHaveBeenCalledTimes(1);
-    expect(mockSetProviderType).toHaveBeenCalledWith(
-      MOCK_MAINNET_CONFIGURATION.type,
-    );
-  });
+  const setupMocks = ({
```

Comment:
optional nit:
```suggestion
  const getMocks = ({
```

Reply:
Recently, I've also come to regard this is as enlightened:
```suggestion
  const makeMocks = ({
```

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/24415/commits/9f416d68ba23367a9dc776248a4a635545fcfa9e